src/main/java/pl/lukakan/showstracker/user/UserController.java

```diff
@@ -0,0 +1,77 @@
+package pl.lukakan.showstracker.user;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import pl.lukakan.showstracker.user.model.UserDto;
+
+@Controller
+@RequestMapping("/user")
+public class UserController {
+
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping("/profile")
+    public String profilePage(Authentication authentication, Model model) {
+        String userName = authentication.getName();
+        UserDto userDto = userService.findUserByName(userName);
+        model.addAttribute("user", userDto);
+        return "user/profile";
+    }
+
+    @GetMapping("/editdata")
+    public String editUserDataForm(Authentication authentication, Model model) {
+        String userName = authentication.getName();
+        UserDto userDto = userService.findUserByName(userName);
+        model.addAttribute("user", userDto);
+        return "user/editdata";
+    }
+
+    @PostMapping("/editdata")
+    public String editUserDataForm(Authentication authentication, UserDto editedUser, RedirectAttributes redirectAttributes) {
+        String userName = authentication.getName();
+        if (!(userName.equals(editedUser.getUserName()))) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Możesz edytować tylko konto: " + userName);
```

Comment:
Zasada jest taka, żeby dawać jak najmniej potencjalnych wskazówek włamywaczom, dlatego jakby ktoś wykradł tokena, to ja bym go nie informował do jakiego on jest konta :P Tak samo jest przy logowaniu - nie podajemy informacji, że login jest zły, albo hasło jest złe. Zawsze piszemy, że są to po prostu są to nieprawidłowe dane do logowania, żeby dużo ciężej było się włamać metodą prób i błędów.

Reply:
hehe głupotka z mojej strony. Dzieki!